R/give_opinions.R

```diff
@@ -1,127 +1,169 @@
+# give_opinions ----------------------------------------------------------------
+
 #' Function giving opinions about a package
 #'
 #' @param pkg_path Path to the package root
 #'
 #' @return A data.frame of opinions
 #' @export
 #'
-give_opinions <- function(pkg_path = getwd()){
-  descr_path <- file.path(pkg_path, "DESCRIPTION")
-  # opinions about DESCRIPTION
-  descr_issues <- give_opinions_desc(descr_path)
-
-  # opinions about README
-  readme <- guess_readme(pkg_path)$readme_path
-  if (is.null(readme)) {
-    readme_issues <- NULL
-  }else{
-    desc_info <- codemeta_description(descr_path)
-    readme_issues <- give_opinions_readme(readme,
-                                          desc_info$identifier)
-  }
+give_opinions <- function(pkg_path = getwd()) {
 
+  # set the path to the description file
+  description_file <- file.path(pkg_path, "DESCRIPTION")
 
-  fixmes <- rbind(descr_issues, readme_issues)
+  # read the description file
+  description <- read_description_if_null(NULL, description_file)
 
-  if(!is.null(fixmes)){
-    descr <- desc::desc(descr_path)
-    fixmes$package <- as.character(descr$get("Package"))
-    return(fixmes)
-  }else{
-    return(NULL)
+  # create a list of issues, each element is a vector of character or NULL
+  fixmes <- do.call(rbind, list(
+
+    # opinions about DESCRIPTION
+    give_opinions_desc(description = description),
+
+    # opinions about README
+    try_to_give_opinions_readme(description_file)
+  ))
+
+  if (! is.null(fixmes)) {
+
+    fixmes$package <- as.character(description$get("Package"))
   }
 
+  fixmes
+}
+
+# read_description_if_null -----------------------------------------------------
+read_description_if_null <- function(description, description_file) {
+
+  if (is.null(description)) {
+
+    desc::desc(description_file)
+
+  } else {
 
+    description
+  }
 }
 
-give_opinions_desc <- function(descr_path){
-  descr <- desc::desc(descr_path)
+# give_opinions_desc -----------------------------------------------------------
+give_opinions_desc <- function(description_file, description = NULL) {
+
+  # read description from description_file if description is NULL
+  description <- read_description_if_null(description, description_file)
```

Comment:
could you add documentation of what description is when not NULL?

Reply:
Bonne idée!